lib/checks/lists/dlitem.js

```diff
@@ -1,2 +1,26 @@
-var parent = axe.commons.dom.getComposedParent(node);
-return parent.nodeName.toUpperCase() === 'DL';
+const parent = axe.commons.dom.getComposedParent(node);
+if (!parent) {
+	return false;
+}
+
+const parentTagName = parent.nodeName.toUpperCase();
+if (parentTagName !== 'DL') {
+	return false;
+}
+
+const parentRole = (parent.getAttribute('role') || '').toLowerCase();
+
+if (!parentRole) {
```

Comment:
With this conditional and the one above, am I reading correctly that this will only match DL elements with a role bolted on? What about other elements?

Reply:
It does tackle them, because we return true. Refer integration tests as well.